tools/prepare-release.js

```diff
@@ -0,0 +1,305 @@
+/**
+ * Prepare a CSS or IDL package release pull request.
+ *
+ * The pull request contains the CSS or IDL diff as description, and bumps the
+ * version patch number.
+ *
+ * If the PR is merged, the "release-package.js" job should run and actually
+ * release the package at the commit on which the pre-release PR was based.
+ *
+ * If the PR is closed, nothing should happen.
+ *
+ * If a PR created by this script is still open, the script (force) updates the
+ * PR.
+ *
+ * Possible TODO improvements:
+ * - Make sure that local clone of the repo is at the same commit as repo on
+ * GitHub (local clone is used to compute the diff). Report an error if that is
+ * not the case.
+ * - Invalidate or re-request review when PR is updated?
+ */
+
+const { Octokit } = require("@octokit/rest");
+const fs = require("fs");
+const path = require("path");
+const os = require("os");
+const { execSync } = require("child_process");
+const rimraf = require("rimraf");
+
+// Repository to process and PR reviewers
+const owner = "w3c";
+const repo = "webref";
+const reviewers = ["dontcallmedom", "foolip", "tidoust"];
+
+// Current package version in npm
+let latestReleasedVersion = "";
+
+
+function btoa(str) {
+  return Buffer.from(str).toString("base64");
+}
+
+
+/**
+ * Compute diff between the released npm package and the contents of the repo
+ *
+ * @function
+ * @param {String} type Package name. One of "css" or "idl"
+ * @return {String} The results of running the diff. An empty string if contents
+ *   match.
+ */
+function computeDiff(type) {
+  // Install @webref package in tmp folder
+  const tmpFolder = fs.mkdtempSync(path.join(os.tmpdir(), "webref-"));
+  execSync(`npm install @webref/${type}`, {
+    cwd: tmpFolder
+  });
+
+  // Extract released version (will be used in the body of the pre-release PR)
+  latestReleasedVersion = require(path.join(tmpFolder, "node_modules", "@webref", type, "package.json")).version;
+
+  // Diff does not take the package.json file into account because "npm install"
+  // adds properties that start with "_" to that file which do not exist in the
+  // repo version of the file.
+  // The "diff" command exits with a non zero status code when there is a diff,
+  // which would throw an exception. Final "echo" command turns that status
+  // code to 0 to avoid the exception.
+  const installedIdl = path.join(tmpFolder, "node_modules", "@webref", type);
+  let diff = execSync(
+    `diff ${installedIdl} packages/${type} --ignore-trailing-space --exclude=package.json --unified=3 || echo -n`,
+    { encoding: "utf8" });
+
+  // Clean up tmp folder
+  try {
+    rimraf.sync(tmpFolder);
+  }
+  catch {
+  }
+
+  // On Windows, the "echo -n" command actually outputs "-n"
+  // Also replace tmp folder name in diff to make it possible to compare diff
```

Comment:
Have you looked for a way to invoke diff that omits the leading parts of the paths in the output? Since `patch -p3` can strip them I half suspect there's a way.

Reply:
I [checked the `diff` manual](https://www.man7.org/linux/man-pages/man1/diff.1.html) and could not find one.